src/processor/operator/aggregate/simple_aggregate.cpp

```diff
@@ -76,27 +172,69 @@ void SimpleAggregate::executeInternal(ExecutionContext* context) {
         for (auto i = 0u; i < aggregateFunctions.size(); i++) {
             auto aggregateFunction = &aggregateFunctions[i];
             if (aggregateFunction->isFunctionDistinct()) {
-                computeDistinctAggregate(distinctHashTables[i].get(), aggregateFunction,
-                    &aggInputs[i], localAggregateStates[i].get(), memoryManager);
+                // Just add distinct value to the hash table. We'll calculate the final hash table
+                // once it's been merged into the global state
+                distinctHashTables[i].get()->insertAggregateValueIfDistinctForGroupByKeys(
+                    std::vector<ValueVector*>{}, aggInputs[i].aggregateVector);
             } else {
                 computeAggregate(aggregateFunction, &aggInputs[i], localAggregateStates[i].get(),
                     memoryManager);
             }
         }
     }
+    if (sharedState->hasDistinct) {
+        for (auto& hashTable : distinctHashTables) {
+            hashTable->mergeAll();
+        }
+    }
     sharedState->combineAggregateStates(localAggregateStates, memoryManager);
 }
 
-void SimpleAggregate::computeDistinctAggregate(AggregateHashTable* distinctHT,
-    function::AggregateFunction* function, AggregateInput* input, function::AggregateState* state,
-    storage::MemoryManager* memoryManager) {
-    auto multiplicity = 1; // Distinct aggregate should ignore multiplicity.
-    if (distinctHT->insertAggregateValueIfDistinctForGroupByKeys(std::vector<ValueVector*>{},
-            input->aggregateVector)) {
-        auto pos = input->aggregateVector->state->getSelVector()[0];
-        if (!input->aggregateVector->isNull(pos)) {
-            function->updatePosState((uint8_t*)state, input->aggregateVector, multiplicity, pos,
-                memoryManager);
+void SimpleAggregateSharedState::finalizeDistinctHashTables(storage::MemoryManager* memoryManager,
```

Comment:
Minor comment on code style, can you reorganize the ordering of function implementations based on their class? Easier to track by just scrolling through the file.

Reply:
I think I missed that one; I was trying to keep them ordered. hash_aggregate.cpp is much worse though.